modules/circe/src/main/scala/pureconfig/module/circe/package.scala

```diff
@@ -0,0 +1,39 @@
+package pureconfig.module
+
+import pureconfig.{ ConfigWriter, ConfigReader }
+import pureconfig.error.{ CannotConvert, FailureReason }
+import com.typesafe.config.{ ConfigValue, ConfigRenderOptions }
+import io.circe.Json
+import io.circe.parser._
+import com.typesafe.config.{ ConfigValueFactory, ConfigValue, ConfigRenderOptions }
+
+import scala.collection.JavaConverters._
+
+package object circe {
+  def cvToJson(cv: ConfigValue): Either[FailureReason, Json] = {
+    val renderOptions = ConfigRenderOptions.concise().setJson(true)
+    val jsonString = cv.render(renderOptions)
+    parse(jsonString) match {
```

Comment:
Point of clarification, as I'm having some difficulty seeing how you avoid using the `circe` parser. Let's say I have `LIST` from `ConfigValue.valueType`. Calling `ConfigValue.unwrapped` on such a structure produces a `List[Object]`, where the children `Object`s have *already* been unwrapped. Digging around the API, it wasn't obvious to me how to avoid this (any of the methods I could find that would have allowed me to handle recursion myself were hidden away in private classes)

Is there something I'm overlooking?

Reply:
I think we need to explicitly cast `ConfigValue` to `ConfigList` when the `valueType` is `LIST`. `ConfigList` then lets you handle each individual element as a `ConfigValue`. It should be reasonably safe to do so in those cases, and it's something that we already resorted to in PureConfig's core.